include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -679,7 +678,127 @@ template <typename Lyt, typename TT>
+    params_with_rejecting_kinks.op_condition          = operational_condition::REJECT_KINKS;
+
+    detail::is_operational_impl<Lyt, TT> p{lyt, spec, params_with_rejecting_kinks};
+
+    std::set<uint64_t> non_op_patterns_due_to_kinks{};
+
+    const auto non_op_patterns = p.determine_non_operational_input_patterns_and_op_status_due_kinks();
+
+    for (const auto& pattern_and_non_op_kinks : non_op_patterns)
+    {
+        if (pattern_and_non_op_kinks.second)
+        {
+            non_op_patterns_due_to_kinks.insert(pattern_and_non_op_kinks.first);
+        }
+    }
+
+    return non_op_patterns_due_to_kinks;
+}
+
+/**
+ * Determines if the layout is `non-operational` due to kinks.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @tparam TT The type of the truth table specifying the layout behavior.
+ * @param lyt The SiDB cell-level layout to be checked.
+ * @param spec Expected Boolean function of the layout given as a multi-output truth table.
+ * @param params Parameters for the `is_operational` algorithm.
+ * @param input_bdl_wire Optional BDL input wires of lyt.
+ * @param output_bdl_wire Optional BDL output wires of lyt.
+ * @return Bool that indicates whether the layout is `non-operational due to kinks. `true` if the layout is not usable
+ * due to kinks, `false` otherwise.
+ *
+ * @note "Non-operational due to kinks" refers to the operational status being exclusively caused to be
+ * `NON_OPERATIONAL` due to kinks.
+ *
+ */
+template <typename Lyt, typename TT>
+[[nodiscard]] bool
+is_non_operational_due_to_kinks(const Lyt& lyt, const std::vector<TT>& spec, const is_operational_params& params = {},
+                                const std::optional<std::vector<bdl_wire<Lyt>>>& input_bdl_wire  = std::nullopt,
+                                const std::optional<std::vector<bdl_wire<Lyt>>>& output_bdl_wire = std::nullopt)
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+    static_assert(kitty::is_truth_table<TT>::value, "TT is not a truth table");
+
+    assert(lyt.num_pis() > 0 && "lyt needs input cells");
+    assert(lyt.num_pos() > 0 && "lyt needs output cells");
+
+    assert(!spec.empty());
+    // all elements in tts must have the same number of variables
+    assert(std::adjacent_find(spec.cbegin(), spec.cend(), [](const auto& a, const auto& b)
+                              { return a.num_vars() != b.num_vars(); }) == spec.cend());
+
+    is_operational_params params_with_rejecting_kinks = params;
+    params_with_rejecting_kinks.op_condition          = operational_condition::REJECT_KINKS;
+
+    if (input_bdl_wire.has_value() && output_bdl_wire.has_value())
+    {
+        detail::is_operational_impl<Lyt, TT> p{lyt, spec, params_with_rejecting_kinks, input_bdl_wire.value(),
+                                               output_bdl_wire.value()};
```

Comment:
hmm, why should we use ``params`` to initiliaze ``p``? 
The reason is that we have to gurantee that the ``operational_condition`` is set to ``REJECT_KINKS``.

Reply:
Yes, but let's say a user does not know how to use the function and uses the default ``params``. The function will not do what it promises because ``accept_kinks`` is set. Do you know what I mean? 
I just want to make it robust.

Reply:
Okay, now I understand. Thanks for clarifying. You may do that of course.